pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1822,21 +1730,23 @@ func (s *AWSCloud) updateInstanceSecurityGroupsForLoadBalancer(lb *elb.LoadBalan
 
 		permissions := []*ec2.IpPermission{permission}
 
-		if add {
-			changed, err := s.ensureSecurityGroupIngress(instanceSecurityGroupId, permissions)
+		if instanceSecurityGroupId != nil && groupId == instanceSecurityGroupId {
+			glog.V(2).Infof("Adding rule for traffic from the load balancer (%s) to instances (%s)", loadBalancerSecurityGroupId, groupId)
```

Comment:
Yep, you're right.  I removed these lines in favor of those inside the sub-functions here.